ownership-chain/pallets/parachain-staking/src/traits.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! traits for parachain-staking
+
+use crate::weights::WeightInfo;
+use frame_support::{dispatch::PostDispatchInfo, pallet_prelude::Weight};
+use sp_runtime::DispatchErrorWithPostInfo;
+
+pub trait OnCollatorPayout<AccountId, Balance> {
+	fn on_collator_payout(
+		for_round: crate::RoundIndex,
+		collator_id: AccountId,
+		amount: Balance,
+	) -> Weight;
+}
+impl<AccountId, Balance> OnCollatorPayout<AccountId, Balance> for () {
```

Comment:
any reason why there's no black space here ?

Reply:
do you mean _blank_ space? if yes, as I said, the code is copied as it is, no modifications from my side.